app/main.py

```diff
@@ -1,2 +1,103 @@
+        index = hash_value % self._capacity
+        for range_index in range(self._capacity):
+            next_index = (index + range_index) % self._capacity
+            node = self.hash_table[next_index]
+            if node is None:
+                self.hash_table[next_index] = Node(key, value, hash_value)
+                self.length += 1
+                if self.length > self._capacity * self._load_factor:
+                    self._resize()
+                return
+            elif node.key == key:
+                new_node = Node(key, value, hash_value)
+                self.hash_table[next_index] = new_node
+                return
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_value = hash(key)
+        index = hash_value % self._capacity
+        for range_index in range(self._capacity):
+            next_index = (index + range_index) % self._capacity
+            node = self.hash_table[next_index]
+            if node is None:
+                break
+            if node.key == key:
+                return node.value
+        raise KeyError(f"Key '{key}' not found")
+
+    def __iter__(self) -> Iterator:
+        for node in self.hash_table:
+            if node is not None:
+                yield node.key
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __delitem__(self, key: Hashable) -> None:
+        hash_value = hash(key)
+        index = hash_value % self._capacity
+        for range_index in range(self._capacity):
+            next_index = (index + range_index) % self._capacity
+            node = self.hash_table[next_index]
+            if node is None:
+                raise KeyError(f"Key '{key}' not found")
+            if node.key == key:
+                self.hash_table[next_index] = None
+                self.length -= 1
+                self._rehash()
+                return
+
+    def _resize(self) -> None:
+        self._capacity = self._capacity * 2
+        old_table = self.hash_table
+        self.hash_table = [None] * self._capacity
+        self.length = 0
+        for node in old_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def _rehash(self) -> None:
+        new_table = Dictionary(self._capacity, self._load_factor)
+        for node in self.hash_table:
+            if node is not None:
+                new_table.__setitem__(node.key, node.value)
+        self.hash_table = new_table.hash_table
+        self.length = new_table.length
+
+    def clear(self) -> None:
+        self.hash_table = [None] * self._capacity
+        self.length = 0
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            value = self[key]
+            del self[key]
+            return value
+        except KeyError:
+            return default
```

Comment:
![image](https://user-images.githubusercontent.com/103201010/221889295-6dd2148d-ffb2-4551-abc5-95b985f089d5.png)
fix this